src/services/window_manager_service/sway/sway_client.c

```diff
@@ -453,6 +454,11 @@ int sway_client_ipc_focus_workspace(int socket_fd, gchar *name) {
     if (!name) {
         return -1;
     }
+
```

Comment:
This won't work in the workspace number is larger then 9 right? 

If you look at sways workspace output in json, does the workplace name match the ID field? If it does it maybe better to just feed the ID of the workspace in this command.

Reply:
It does work for workspaces over 9, after checking if the first digit is a number, it then passes the whole name string to the modified command with the number parameter added.

Running the unpatched version, when I click the workspace button in the bar it creates a new workspace with the exact same name, but a different ID.  I am using the <num>:<name> format, with the name representing the windows running on the workspace. 

The patch here does fix that problem, but I'm not sure if it creates any others, I haven't found any issues in my testing so far.

Reply:
Ah yes i see. Tho what I'm trying to gather is if passing: https://github.com/ldelossa/way-shell/blob/main/src/services/window_manager_service/window_manager_service.h#L41

is s bit better then the isdigit check? I'd need to see the results of "get_workspaces" command in JSON after your script creates a few workspaces

Reply:
Okay, I confirmed, the ID's do not match up with the workspaces when I provide it to 'workspace number x'